lib/src/util/component_usage.dart

```diff
@@ -564,29 +570,51 @@ FluentComponentUsage? getComponentUsage(InvocationExpression node) {
     builder = functionExpression;
   }
 
-  bool isComponent;
```

Comment:
I changed this variable to `final` so that the analyzer would enforce that it's assigned exactly one time (and isn't ever uninitialized or overwritten), which I felt made it easier to ensure there weren't any unhandled cases in the relatively complex if-else conditions below.